uint8 pin
float32 percent
---
uint8 result
